### ISSUE TITLE
batch_size must be small than data pool_size?why?

### ISSUE THREAD

**Issue by wangsouc** [PaddlePaddle/Paddle#2109]
batch_size = 128, data pool_size = 500,  train well
batch_size = 128, data pool_size = 50, train hang 

batch_size must be small than data pool_size? why? how to set a better data pool_size?

**Comment by lcy-seso** [comment]
Data feeding interface in PaddlePaddle is implemented by python generator, that the user only needs to focus on how to generate one training sample from the original data file. At the meanwhile, how well training data are shuffled often affected the learning performance.

In PaddlePaddle `pool_size ` controls a minimum amount of training samples shuffled. Roughly speaking, PaddlePaddle maintains a pool with the size indicated by `pool_size`.

Before starting training, PaddlePaddle fills the pool and shuffle it first, and then feeds the shuffled data to the training process batch by batch.

If you set a small pool size, the training data will not be well shuffled, which usually leads to a decline in learning performance according to my experience. A large pool size requires much more memory, but if you have enough CPU memory, I suggest a large pool size guarantee the randomness of training samples.
